src/main/common/maths.c

```diff
@@ -130,6 +130,18 @@ int32_t applyDeadband(int32_t value, int32_t deadband)
     return value;
 }
 
+float fapplyDeadband(float value, float deadband)
+{
+    if (fabsf(value) < deadband) {
+        value = 0;
+    } else if (value > 0) {
+        value -= deadband;
+    } else if (value < 0) {
+        value += deadband;
```

Comment:
Immutability looks not the best:
I would rewrite it with 3 returns:
```cpp
if (fabsf(value) < deadband) {
   return 0;
} else if (value > 0) {
   return value - deadband;
} else {
   return value + deadband;
}
```

Reply:
I copied it from the implementation of `applyDeadband()`. Both of these should be implemented in the same way. I wasn't sure if there was any significance to returning `value` in the case that `value == 0 && deadband <= 0`, but I did not want to risk making them non-interchangeable.

Can you propose a better implementation of `applyDeadband()` and `fapplyDeadband()` in a separate pull request?

Reply:
```cpp
if (fabsf(value) < deadband) {
   return 0;
} else if (value > 0) {
   return value - deadband;
} else if (value < 0) {
   return value + deadband;
}

return value;
```

Reply:
Initially I just wanted the mutability to be removed